src/main/java/org/jabref/model/openoffice/util/OOResult.java

```diff
@@ -0,0 +1,128 @@
+package org.jabref.model.openoffice.util;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/*
+ * error cannot be null
+ * result cannot be null
+ *
+ * Void is not allowed for R, use OOVoidResult instead.
+ *
+ * Out of `isPresent()` and `isError()` exactly one is true.
+ */
+public class OOResult<R, E> {
+    private final Optional<R> result;
+    private final Optional<E> error;
+
+    /**
+     * Exactly one of the arguments should be Optional.empty()
+     *
+     * @param result
+     * @param error
+     */
+    private OOResult(Optional<R> result, Optional<E> error) {
```

Comment:
Please, never ever use `Optional` as method parameter.

The most prominent way to fix is to offer different methods.

You already have these methods, thus just remove this method.

Reply:
Stuck at ambiguity problem, see below. https://github.com/JabRef/jabref/pull/7787#discussion_r645172827
Could probably use constructor with no arguments. Would that be better?

Reply:
With following comment

> When not using Optionals as method parameters, the following code should do it:

I meant:

Please do not use `Optionals` as parameter. Please use my suggested code.

Reply:
Reading further: This seems to be impossible. Then, just leave as is.

Reply:
Or use `ofError` and `ofResult`.

Reply:
```
return OOResult.ok(cursor);
return OOResult.error(message);
```
vs
```
return OOResult.ofResult(cursor);
return OOResult.ofError(message);
```
The first pair seems more suggestive of the important part: success or failure.
The part that we are constructing something (`of`) is just background, not worth spelling out.